actors/builtin/storage_miner/storage_miner_actor.go

```diff
@@ -167,27 +167,19 @@ func (a *StorageMinerActor) PreCommitSector(rt Runtime, info SectorPreCommitInfo
 	depositReq := cidx.StorageMining_PreCommitDeposit(st.Info.SectorSize, info.Expiration)
 	vmr.RT_ConfirmFundsReceiptOrAbort_RefundRemainder(rt, depositReq)
 
-	// Verify deals with StorageMarketActor; abort if this fails.
-	// (Note: committed-capacity sectors contain no deals, so in that case verification will pass trivially.)
-	_, code := rt.Send(
-		builtin.StorageMarketActorAddr,
-		builtin.Method_StorageMarketActor_OnMinerSectorPreCommit_VerifyDealsOrAbort,
-		serde.MustSerializeParams(
-			info.DealIDs,
-			info,
-		),
-		abi.NewTokenAmount(0),
-	)
-	vmr.RequireSuccess(rt, code, "failed to verify deals")
+	if info.SealEpoch >= (rt.CurrEpoch() + indices.StorageMining_SpcLookbackSeal()) {
```

Comment:
What is `SealEpoch`? The epoch of pre-commit, or the deadline for provecommit (it's undocumented)? If the former, this condition can never succeed so is probably a bug. If the latter, can we name it as such? The error message below is very opaque, especially since the numbers have no labels. Try something like "seal epoch (%d) must be before current epoch (%d) + lookback (%d) = %d"

"Seal" is also not a word that's really in the vocab of the actors. Use pre-commit or provecommit. Does seal == prove? seal == commit?

Reply:
sorry we tried to document it with Porcu in https://github.com/filecoin-project/specs-actors/blob/master/actors/abi/sector.go#L80.

`SealEpoch` is used to tie the sealing to the chain as a whole, so it is drawn from finality back at pre-commit (more details in spec). It matters on verification since it accounts for how long sealing actually takes. I'm not quitee sure what is happening in this function per se, but sealEpoch should be `currEpoch - spcLookbackSeal` when miner starts a precommit, thereafter the actor should verify it per (see https://github.com/filecoin-project/specs-actors/pull/29/files#diff-6c14f5a2f4e3689416a73d535d06d2b3R735-R740).

Regarding the nomenclature I think dig's terminology is used here cc @porcuquine

Reply:
Removing checks on SealEpoch from this PR for now. @sternhenri could you include a PR here for the right checks next week? thanks